src/modules/job-manager/alloc.c

```diff
@@ -50,9 +50,34 @@ struct alloc {
     flux_watcher_t *idle;
     unsigned int alloc_pending_count; // for mode=single, max of 1
     unsigned int free_pending_count;
+    struct job *pending_job; // for mode=single
```

Comment:
I haven't looked at the code, so take my comment with a grain of salt, but it feels a bit odd to treat the "single" mode special here and have a single pending job pointer saved.

Aren't all the jobs with outstanding alloc requests at the front of the queue, so `pending_job` in single alloc mode would just be `zlistx_first (alloc->queue)` if `alloc->alloc_pending_count != 0`?

Reply:
in the current code once a job is sent to the scheduler its taken off the alloc queue.

I could just save off the job id or something like that and re-look up the pointer later?

Reply:
Ah, no, then what you are doing makes sense.. though it seems odd that pending allocs aren't on the alloc queue, but probably I don't understand the job-manager internals at the moment.

My overall point was that it would be nice if we didn't have to special-case the "single" alloc mode.. but maybe that isn't possible ATM.,

Reply:
OR could it just be inserted back in the alloc queue in the new priority order?

Reply:
> OR could it just be inserted back in the alloc queue in the new priority order?

you mean keep the job in the alloc->queue and not remove it when the alloc request is sent?  Then everytime a job is submitted or job urgency is changed, make sure that the head of `alloc->queue` hasn't changed?

i guess that would work, although I'm not sure if its a huge net win on code.  Because if I'm not saving a pointer to the job that has had its alloc request sent, I'll just have to save it before inserting a new job, or search for the `job->alloc_pending` job.

Reply:
oops I misunderstood - that suggestion was a bit off!  Sorry.

I think you'll want to replace this variable with a `zlist_t *alloc_pending` when you implement `sched.prioritize` so that you won't have to walk the entire active job to construct the request.    Does that make sense?

I'm not sure if it makes sense to just do that now?  Up to you.

Reply:
yeah, makes sense.  Lets hold off until we start work on that PR.